src/fund/components/FundCardPaymentMethodDropdown.tsx

```diff
@@ -0,0 +1,92 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { background, border, cn } from '../../styles/theme';
+
+import { useOutsideClick } from '@/ui-react/internal/hooks/useOutsideClick';
+import type {
+  FundCardPaymentMethodDropdownPropsReact,
+  PaymentMethodReact,
+} from '../types';
+import { FundCardPaymentMethodSelectRow } from './FundCardPaymentMethodSelectRow';
+import { FundCardPaymentMethodSelectorToggle } from './FundCardPaymentMethodSelectorToggle';
+import { useFundContext } from './FundCardProvider';
+
+export function FundCardPaymentMethodDropdown({
+  paymentMethods,
+}: FundCardPaymentMethodDropdownPropsReact) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { selectedPaymentMethod, setSelectedPaymentMethod } = useFundContext();
+
+  const handlePaymentMethodSelect = useCallback(
+    (paymentMethod: PaymentMethodReact) => {
+      setSelectedPaymentMethod(paymentMethod);
+      setIsOpen(false);
+    },
+    [setSelectedPaymentMethod],
+  );
+
+  const handleToggle = useCallback(() => {
+    setIsOpen(!isOpen);
+  }, [isOpen]);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const dropdownContainerRef = useRef<HTMLDivElement>(null);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: This useEffect is only called once
+  useEffect(() => {
+    setSelectedPaymentMethod(paymentMethods[0]);
+  }, []);
+
+  useOutsideClick(dropdownContainerRef, () => {
+    if (isOpen) {
+      setIsOpen(false);
+    }
+  });
+
+  const handleEscKeyPress = useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>) => {
+      if (event.key === 'Escape') {
+        setIsOpen(false);
+      }
+    },
+    [],
+  );
+
+  return (
+    <div
+      className="relative py-4"
+      ref={dropdownContainerRef}
+      data-testid="ockFundCardPaymentMethodDropdownContainer"
+      onKeyUp={handleEscKeyPress}
+    >
+      <FundCardPaymentMethodSelectorToggle
+        ref={buttonRef}
+        onClick={handleToggle}
+        isOpen={isOpen}
+        paymentMethod={selectedPaymentMethod || paymentMethods[0]}
```

Comment:
why is this needed if you are doing this in the useEffect, or can you remove the useEffect and do this instead?